src/Core/src/ImageSources/ImageSourceServiceResultManager.cs

```diff
@@ -24,11 +43,31 @@ public CancellationToken BeginLoad()
 			return Token;
 		}
 
-		public void CompleteLoad<T>(IImageSourceServiceResult<T>? result)
+		public void CompleteLoad<T>(IImageSourceServiceResult<T>? result, PlatformView uiContext)
 		{
 			CompleteLoad((IDisposable?)result);
 
 			IsResolutionDependent = result?.IsResolutionDependent ?? false;
+
+#if IOS || MACCATALYST || WINDOWS
+			CurrentResolution = uiContext.GetDisplayDensity();
+#endif
+		}
+
+		public bool RequiresReload(PlatformView uiContext)
```

Comment:
I wonder if we should have an overload that takes a IMauiContext so we can get the window density instead of the view stuff. I am thinking in Windows where a view does not have a Window property...

Reply:
Will address in a future PR.